pkg/kademlia/routing.go

```diff
@@ -112,10 +112,18 @@ func (rt *RoutingTable) GetBucket(id string) (bucket dht.Bucket, ok bool) {
 // GetBuckets retrieves all buckets from the local node
 func (rt *RoutingTable) GetBuckets() (k []dht.Bucket, err error) {
 	bs := []dht.Bucket{}
-	kbuckets, err := rt.kadBucketDB.List(nil, 0)
+
+	kbucketsItems, _, err := rt.kadBucketDB.List(storage.ListOptions{
+		Start: nil,
+		Limit: 0,
+	})
+
 	if err != nil {
 		return bs, RoutingErr.New("could not get bucket ids %s", err)
 	}
+
+	kbuckets := kbucketsItems.GetKeys()
+
 	for _, v := range kbuckets {
```

Comment:
this for loop needs to be changed to actually use the pagination kadBucketDB.List now supports

Reply:
```
listOpts := storage.ListOptions{Limit: 100}
for {
  kbucketItems, more, err := rt.kadBucketDB.List(listOpts)
  if err != nil { 
    return bs, err
  }
  for _, v := range kbucketsItems.GetKeys() {
    ...
  }
  if !more { 
    break
  }
  listOpts.Start = kbucketItems[len(kbucketItems)-1]
}
```